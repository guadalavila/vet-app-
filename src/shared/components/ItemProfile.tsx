import React from 'react';
import { StyleSheet, View } from 'react-native';
import { size } from '~shared/utils/size';
import { typography } from '~shared/utils/typography';
import CustomText from './CustomText';
import Separator from './Separator';

interface IItemProfile {
    title: string;
    data: string;
}

const ItemProfile = ({ title, data }: IItemProfile) => {
    return (
        <View testID='ItemProfile' style={styles.info}>
            <CustomText style={styles.title}>{title}</CustomText>
            <CustomText style={styles.margin}>{data}</CustomText>
            <Separator testID='separator' />
        </View>
    );
};

export default ItemProfile;

const styles = StyleSheet.create({
    info: {
        marginHorizontal: size.XL,
        marginVertical: size.M,
    },
    margin: {
        marginVertical: size.XS,
    },
    title: {
        fontWeight: '500',
        fontSize: typography.size.S,
    },
});
