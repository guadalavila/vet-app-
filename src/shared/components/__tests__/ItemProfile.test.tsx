import { render, screen } from '@testing-library/react-native';
import React from 'react';
import ItemProfile from '../ItemProfile';

describe('ItemProfile', () => {
    test('should match snapshot', () => {
        const container = render(<ItemProfile title='Hello' data='World!' />);
        expect(container).toMatchSnapshot();
    });

    test('should render component', () => {
        render(<ItemProfile title='Hello' data='World!' />);
        expect(screen.getByTestId('ItemProfile')).toBeTruthy();
    });

    test('should render title', () => {
        render(<ItemProfile title='Hello' data='World!' />);
        expect(screen.getByText('Hello')).toBeTruthy();
        expect(screen.getByText('World!')).toBeTruthy();
    });

    test('should render Separator', () => {
        render(<ItemProfile title='Hello' data='World!' />);
        expect(screen.getByTestId('separator')).toBeTruthy();
    });
});
