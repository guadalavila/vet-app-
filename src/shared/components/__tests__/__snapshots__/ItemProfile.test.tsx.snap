// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemProfile should match snapshot 1`] = `
<View
  style={
    Object {
      "marginHorizontal": 10,
      "marginVertical": 6,
    }
  }
  testID="ItemProfile"
>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Object {
          "fontSize": 16,
          "fontWeight": "500",
        },
      ]
    }
  >
    Hello
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#11100F",
          "fontSize": 16,
        },
        Object {
          "fontSize": 14,
          "marginVertical": 3,
        },
      ]
    }
  >
    World!
  </Text>
  <View
    style={
      Array [
        Object {
          "borderWidth": 0.4,
          "marginVertical": 6,
        },
        Object {
          "borderColor": "#F1f1f2",
        },
      ]
    }
    testID="separator"
  />
</View>
`;
